crates/gpui/src/platform/linux/gtk_utils.rs

```diff
@@ -0,0 +1,276 @@
+/// Based on rfd crate's gtk backend. Useful to show GTK Widgets asynchronously and by blocking.
```

Comment:
Like I mentioned. The `gtk_utils.rs` file is almost entirely based on rfd crate's code. I combined both [gtk_future.rs](https://github.com/PolyMeilex/rfd/blob/master/src/backend/gtk3/gtk_future.rs) and [utils.rs](https://github.com/PolyMeilex/rfd/blob/master/src/backend/gtk3/utils.rs) files into single `gtk_utils.rs` file for use here. While I understand the gist of what this code does, it's bit overwhelming for a rust newcomer like me :face_with_spiral_eyes: